--- ping6.c.orig	2002-09-20 17:08:11.000000000 +0200
+++ ping6.c	2004-01-27 03:52:42.704942760 +0100
@@ -788,38 +788,40 @@
 		printf("Destination unreachable: ");
 		switch (code) {
 		case ICMPV6_NOROUTE:
-			printf("No route");
+			printf("No route\n");
 			break;
 		case ICMPV6_ADM_PROHIBITED:
-			printf("Administratively prohibited");
+			printf("Administratively prohibited\n");
 			break;
 		case ICMPV6_NOT_NEIGHBOUR:
-			printf("Not neighbour");
+			printf("Not neighbour\n");
 			break;
 		case ICMPV6_ADDR_UNREACH:
-			printf("Address unreachable");
+			printf("Address unreachable\n");
 			break;
 		case ICMPV6_PORT_UNREACH:
-			printf("Port unreachable");
+			printf("Port unreachable\n");
 			break;
 		default:	
-			printf("Unknown code %d", code);
+			printf("Unknown code %d\n", code);
 			break;
 		}
 		break;
 	case ICMPV6_PKT_TOOBIG:
 		printf("Packet too big: mtu=%u", info);
 		if (code)
-			printf(", code=%d", code);
+			printf(", code=%d\n", code);
+		else
+			putchar('\n');
 		break;
 	case ICMPV6_TIME_EXCEED:
 		printf("Time exceeded: ");
 		if (code == ICMPV6_EXC_HOPLIMIT)
-			printf("Hop limit");
+			printf("Hop limit\n");
 		else if (code == ICMPV6_EXC_FRAGTIME)
-			printf("Defragmentation failure");
+			printf("Defragmentation failure\n");
 		else
-			printf("code %d", code);
+			printf("code %d\n", code);
 		break;
 	case ICMPV6_PARAMPROB:
 		printf("Parameter problem: ");
@@ -831,25 +833,25 @@
 			printf("Unknown option ");
 		else
 			printf("code %d ", code);
-		printf ("at %u", info);
+		printf ("at %u\n", info);
 		break;
 	case ICMPV6_ECHO_REQUEST:
-		printf("Echo request");
+		printf("Echo request\n");
 		break;
 	case ICMPV6_ECHO_REPLY:
-		printf("Echo reply");
+		printf("Echo reply\n");
 		break;
 	case ICMPV6_MGM_QUERY:
-		printf("MLD Query");
+		printf("MLD Query\n");
 		break;
 	case ICMPV6_MGM_REPORT:
-		printf("MLD Report");
+		printf("MLD Report\n");
 		break;
 	case ICMPV6_MGM_REDUCTION:
-		printf("MLD Reduction");
+		printf("MLD Reduction\n");
 		break;
 	default:
-		printf("unknown icmp type");
+		printf("unknown icmp type\n");
 		
 	}
 	return 0;
