diff -Nru iputils.orig/ping_common.c iputils/ping_common.c
--- iputils.orig/ping_common.c	Fri Sep 20 18:02:32 2002
+++ iputils/ping_common.c	Wed Jan 22 14:53:48 2003
@@ -39,16 +39,16 @@
 int working_recverr;
 
 /* timing */
-int timing;			/* flag to do timing */
-long tmin = LONG_MAX;		/* minimum round trip time */
-long tmax;			/* maximum round trip time */
+unsigned long long timing;			/* flag to do timing */
+unsigned long long tmin = LONG_MAX;		/* minimum round trip time */
+unsigned long long tmax;			/* maximum round trip time */
 /* Message for rpm maintainers: have _shame_. If you want
  * to fix something send the patch to me for sanity checking.
  * "sparcfix" patch is a complete non-sense, apparenly the person
  * prepared it was stoned.
  */
-long long tsum;			/* sum of all times, for doing average */
-long long tsum2;
+unsigned long long tsum;			/* sum of all times, for doing average */
+unsigned long long tsum2;
 int  pipesize = -1;
 
 int datalen = DEFDATALEN;
@@ -809,10 +809,10 @@
 		tmdev = llsqrt(tsum2 - tsum * tsum);
 
 		printf("rtt min/avg/max/mdev = %ld.%03ld/%lu.%03ld/%ld.%03ld/%ld.%03ld ms",
-		       tmin/1000, tmin%1000,
+		       (long)tmin/1000, (long)tmin%1000,
 		       (unsigned long)(tsum/1000), (long)(tsum%1000),
-		       tmax/1000, tmax%1000,
-		       tmdev/1000, tmdev%1000
+		       (long)tmax/1000, (long)tmax%1000,
+		       (long)tmdev/1000, (long)tmdev%1000
 		       );
 	}
 	if (pipesize > 1)
@@ -843,10 +843,10 @@
 		tavg = tsum / (nreceived + nrepeats);
 
 		fprintf(stderr, ", min/avg/ewma/max = %ld.%03ld/%lu.%03ld/%d.%03d/%ld.%03ld ms",
-		       tmin/1000, tmin%1000,
+		       (long)tmin/1000, (long)tmin%1000,
 		       tavg/1000, tavg%1000,
 		       rtt/8000, (rtt/8)%1000,
-		       tmax/1000, tmax%1000
+		       (long)tmax/1000, (long)tmax%1000
 		       );
 	}
 	fprintf(stderr, "\n");
diff -Nru iputils.orig/ping_common.h iputils/ping_common.h
--- iputils.orig/ping_common.h	Fri Sep 20 17:08:11 2002
+++ iputils/ping_common.h	Wed Jan 22 14:52:08 2003
@@ -103,11 +103,11 @@
 
 
 /* timing */
-extern int timing;			/* flag to do timing */
-extern long tmin;			/* minimum round trip time */
-extern long tmax;			/* maximum round trip time */
-extern long long tsum;			/* sum of all times, for doing average */
-extern long long tsum2;
+extern unsigned long long timing;		/* flag to do timing */
+extern unsigned long long tmin;			/* minimum round trip time */
+extern unsigned long long tmax;			/* maximum round trip time */
+extern unsigned long long tsum;			/* sum of all times, for doing average */
+extern unsigned long long tsum2;
 extern int rtt;
 extern __u16 acked;
 extern int pipesize;
