diff -urNp iputils-s20100418/arping.c iputils-s20100418.bcast/arping.c
--- iputils-s20100418/arping.c	2010-04-18 06:45:45.000000000 +0200
+++ iputils-s20100418.bcast/arping.c	2010-08-19 11:52:13.000000000 +0200
@@ -336,7 +336,7 @@ void set_device_broadcast(char *device, 
 	}
 
 	for (p = ba, ch = 0; p < ba + balen; p++, ch += 3)
-		*p++ = strtoul(brdcast->value + ch * 3, NULL, 16);
+		*p = strtoul(brdcast->value + ch, NULL, 16);
 
 	return;
 }
