--- iputils/Makefile.rh7	Thu Jan 24 18:27:37 2002
+++ iputils/Makefile	Tue Feb 19 16:58:11 2002
@@ -25,13 +25,13 @@
 # What a pity, all new gccs are buggy and -Werror does not work. Sigh.
 #CCOPT=-D_GNU_SOURCE -O2 -Wstrict-prototypes -Wall -g -Werror
 CCOPT=-D_GNU_SOURCE -O2 -Wstrict-prototypes -Wall -g
-CFLAGS=$(CCOPT) $(GLIBCFIX) -I$(KERNEL_INCLUDE) -I../include $(DEFINES) 
+CFLAGS=$(CCOPT) $(GLIBCFIX) -I../include $(DEFINES) 
 
 IPV4_TARGETS=tracepath ping clockdiff rdisc arping tftpd rarpd
 IPV6_TARGETS=tracepath6 traceroute6 ping6
 TARGETS=$(IPV4_TARGETS) $(IPV6_TARGETS)
 
-all: check-kernel $(TARGETS)
+all: $(TARGETS)
 
 
 tftpd: tftpd.o tftpsubs.o
--- iputils/doc/Makefile.foo	Thu Apr 18 17:34:58 2002
+++ iputils/doc/Makefile	Thu Apr 18 17:35:02 2002
@@ -2,7 +2,7 @@
 HTMLFILES=$(subst .sgml,.html,$(shell echo *.sgml))
 MANFILES=$(subst .sgml,.8,$(shell echo *.sgml))
 
-all: html
+all:
 
 html: $(HTMLFILES) iputils.html index.html
 
