--- iputils-s20100418/Makefile~	2010-08-15 21:40:16.369143350 +0200
+++ iputils-s20100418/Makefile	2010-08-15 21:40:55.575806109 +0200
@@ -31,8 +31,8 @@
 	$(LINK.o) $^ $(LDLIBS) -o $@ -Wl,-z,now -Wl,-z,relro
 ping: ping.o ping_common.o
 	$(LINK.o) $^ $(LDLIBS) -o $@ -Wl,-z,now -Wl,-z,relro
-ping6: ping6.o ping_common.o -lresolv -lcrypto
-	$(LINK.o) $^ $(LDLIBS) -o $@ -Wl,-z,now -Wl,-z,relro
+ping6: ping6.o ping_common.o
+	$(LINK.o) $^ $(LDLIBS) -o $@ -lresolv -lcrypto -Wl,-z,now -Wl,-z,relro
 traceroute6: traceroute6.o
 	$(LINK.o) $^ $(LDLIBS) -o $@ -Wl,-z,now -Wl,-z,relro
 ping.o ping6.o ping_common.o: ping_common.h
