diff -urN iputils.org/arping.c iputils/arping.c
--- iputils.org/arping.c	Tue Nov 14 18:36:51 2000
+++ iputils/arping.c	Tue Nov 14 18:57:08 2000
@@ -17,8 +17,10 @@
 #include <sys/time.h>
 #include <sys/signal.h>
 #include <sys/ioctl.h>
-#include <linux/if.h>
-#include <linux/if_arp.h>
+#include <net/if.h>
+#include <net/if_arp.h>
+#include <netpacket/packet.h>
+#include <linux/if_ether.h>
 #include <sys/uio.h>
 
 #include <netdb.h>
diff -urN iputils.org/ping6.c iputils/ping6.c
--- iputils.org/ping6.c	Tue Nov 14 18:36:51 2000
+++ iputils/ping6.c	Tue Nov 14 18:57:37 2000
@@ -72,6 +72,7 @@
 #include <string.h>
 #include <sys/param.h>
 #include <sys/socket.h>
+#include <linux/config.h>
 #include <linux/sockios.h>
 #include <sys/file.h>
 #include <sys/time.h>
