--- iputils-s20121114/doc/ping.sgml.orig	2012-11-14 14:10:12.000000000 +0100
+++ iputils-s20121114/doc/ping.sgml	2012-11-15 19:41:09.023053608 +0100
@@ -130,6 +130,7 @@
   <listitem><para>
 Show help.
   </para></listitem>
+ </varlistentry>
  <varlistentry>
   <term><option>-i <replaceable/interval/</option></term>
   <listitem><para>
--- iputils-s20121114/doc/tracepath.sgml.orig	2012-11-14 14:10:12.000000000 +0100
+++ iputils-s20121114/doc/tracepath.sgml	2012-11-15 19:42:04.633052399 +0100
@@ -72,6 +72,7 @@
   <listitem><para>
 Sets the initial destination port to use.
   </para></listitem>
+ </varlistentry>
 </variablelist>
 </refsect1>
 
