--- Makefile~	2004-05-02 20:38:11.000000000 +0000
+++ Makefile	2004-05-02 20:40:11.000000000 +0000
@@ -4,7 +4,7 @@
 DEFINES= 
 
 #options if you have a bind>=4.9.4 libresolv (or, maybe, glibc)
-LDLIBS=-lresolv
+LDLIBS=
 ADDLIB=
 
 
@@ -21,7 +21,7 @@
 IPV6_TARGETS=tracepath6 traceroute6 ping6
 TARGETS=$(IPV4_TARGETS) $(IPV6_TARGETS)
 
-all: subdirs $(TARGETS)
+all: $(TARGETS)
 
 
 tftpd: tftpd.o tftpsubs.o
@@ -35,17 +35,6 @@
 rdisc_srv.o: rdisc.c
 	$(CC) $(CFLAGS) -DRDISC_SERVER -o rdisc_srv.o rdisc.c
 
-subdirs:
-ifneq ($(SUBDIRS),)
-	@set -e; \
-	for i in $(SUBDIRS); \
-	do $(MAKE) -C $$i; done; \
-	cd racoon; ./configure; make
-endif
-
-
-modules: check-kernel
-	$(MAKE) KERNEL_INCLUDE=$(KERNEL_INCLUDE) -C Modules
 
 man:
 	$(MAKE) -C doc man
